### ISSUE TITLE
Convert Python Package from Pandas to Polars

### ISSUE THREAD

**Issue by jjacobson95** [PNNL-CompBio/coderdata#159]
@sgosline We've briefly spoken about this before and it came up after my comp bio presentation as well. I'd like to do this before addressing #111.

Currently the package is build from Python, converting it all to Polars will speed up operations and make it less susceptible to memory issues as we continue to add more data. Even now, I could see users on low memory devices potentially having issues.

This update should address the following 7 files:
- Data Loader Script (loader.py)
- Data Exporation Tutorial & HTML version in docs
- Drug Exploration Tutorial & HTML version in docs
- Deep Learning Tutorial & HTML version in docs

**Comment by sgosline** [comment]
This seems like a good move. It'd be nice to move all the build scripts to polars as well, but that will break a lot.

**Comment by jjacobson95** [comment]
Great, I fully agree with that too.

**Comment by sgosline** [comment]
Just to be explicit (I thought it was assumed), we should make sure that the forward facing code is still in pandas, so that users dont have to learn polars to use the package.

**Comment by jjacobson95** [comment]
I thought the goal was to migrate the package from pandas to polars?  Pandas is slower and has memory issues. If users have to learn one of the two, they might as well learn the one that is more optimized for our data?

Everything in the package that uses pandas is user-facing.

**Comment by jjacobson95** [comment]
As this code is nearly complete, we could have two versions if we wanted? - DatasetLoader_pd() and DatasetLoader_pl()? 

But having two versions would also increase time for future code development such as the upcoming DataTypeLoader.

**Comment by sgosline** [comment]
no, that'd be a nightmare.   Just add a flag called 'use_polars' and set the default to False. HOw you imagine this working when #111 is complete?

**Comment by jjacobson95** [comment]
Oh yeah, thats a much better solution (using a flag). For #111, I assume we should also create a pandas and polars version as well (with the same flag option)?
